mlflow/pyfunc/__init__.py

```diff
@@ -2326,8 +2346,6 @@ def predict(model_input: List[str]) -> List[str]:
         )
         raise MlflowException(message=msg, error_code=INVALID_PARAMETER_VALUE)
 
-    _validate_and_prepare_target_save_path(path)
```

Comment:
Curious on why this was moved up? Not blocking, just curious

Reply:
if the python model is a path to a notebook, we are saving the contents of the notebook into `path`. if this is lower, then it will fail since the folder is (expectedly) not empty